### ISSUE TITLE
Password protected public link shares : time to improve password protection UI to avoid a trap that many users fall into

### ISSUE THREAD

**Issue by Jerome-Herbinet** [nextcloud/server#46988]
For a long time now, the password protection UI has been the cause of failure for many users, whether "gifted" or not. 
Whether the password is optional (activated manually by clicking on the checkbox), proposed by default or even imposed, one step is often overlooked: recovering the password before it is hashed and irrecoverable. 
I see two improvements: 
- Add a sentence "For security reasons, this password will only be shown once" (unlike screenshots, remember to include the comma). I didn't invent this sentence. It's already used elsewhere, to generate application passwords. In my opinion, this phase should perhaps be bolded, or preceded by a small warning icon.
- I find the password recovery process rather tedious. First of all, whether the password is proposed by default or not, or even imposed, there is no copy button/icon, which forces the user to select the password, then do a CTRL+C or right-click "copy". It's even worse when the password is neither proposed by default nor imposed, as it appears hidden and you have to click on the eye to copy it. A copy button must therefore be added.

But this isn't perfect, because in the case of a password-protected public link offered by default or forced, there are 2 pieces of information to copy and paste (in one or even two separate e-mails, for example): the URL and the password. So perhaps, when the password is proposed by default or forced, you shouldn't put a copy button as I've just suggested, but copy automatically to the clipboard after creating the share. Example: "Share URL: {share url} - Share password: {share password}". In the case of shares where the password is completely optional (not proposed by default, and certainly not imposed), we would still have the copy button in the field, after manual activation of password protection.

![2024-08-02_15-44](https://github.com/user-attachments/assets/2f284432-9a1b-4a91-a500-304679b7e448)
![2024-08-02_15-37](https://github.com/user-attachments/assets/fefb7c6c-71de-4f7b-8a0b-054d8892cf1b)

### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by Jerome-Herbinet** [comment]
@nextcloud/designers you may hate me, but I love you 😇

**Comment by Mika-6-9** [comment]
Hi.
For me, if there were to be changes in the code, it would be better to use the current share link copy button (see below) and transform it so that when you click on it, you copy something like:
"_The share link is: https://xxxxxxx
and the password to access the link is: xxxxx_"
We would just have to copy this into an email.
This would avoid a copy/paste (one for the link and one for the password).
Another option: clicking on this link opens a "mailto" that puts this text in the body of the email.

![image](https://github.com/user-attachments/assets/0818f85f-4ca8-4042-8e12-71819adc756b)

**Comment by jancborchardt** [comment]
Yeah, I think both the suggestions by @Jerome-Herbinet and @Mika-6-9 make sense. The copy icon in the password field for when you are in the flow, or decide to password-protect an existing share, and the inclusion in the copied text to make it easier for quickly copying everything.

@Jerome-Herbinet @Mika-6-9 or anyone else in here, would you like to contribute this or parts of it as a pull request?

**Comment by Jerome-Herbinet** [comment]
> Yeah, I think both the suggestions by @Jerome-Herbinet and @Mika-6-9 make sense. The copy icon in the password field for when you are in the flow, or decide to password-protect an existing share, and the inclusion in the copied text to make it easier for quickly copying everything.
> 
> @Jerome-Herbinet @Mika-6-9 or anyone else in here, would you like to contribute this or parts of it as a pull request?

I'm afraid I'm running out of skills in this context.

**Comment by joshtrichards** [comment]
Related: #46914
